### PR TITLE
Bug 1097090 - jobs endpoint refactoring

### PR DESCRIPTION
**Pull request by maurodoglio**
Tests are passing. This is the service part of mozilla/treeherder-ui#340

### CODE REVIEW

**Comment by lightsofapollo** [comment]
ah nice! this may fix several of the issues I have run into

File: treeherder/webapp/api/jobs.py [hunk 1]
```diff
@@ -54,22 +54,37 @@ def retrieve(self, request, project, jm, pk=None):
     def list(self, request, project, jm):
         """
         GET method implementation for list view
-
+        Optional paramters (default):
+        - offset (0)
+        - count (10)
+        - full (false)
```
Comment (camd) [inline treeherder/webapp/api/jobs.py]: isn't the ``full`` param removed now?  We could remove it from this list of optional fields.

**Comment by camd** [comment]
@maurodoglio with this change, seems like we should remove the ``get_result_set_job_list`` function because it's just dead code:
https://github.com/mozilla/treeherder-service/blob/bug-1097090-jobs-endpoint-refactoring/treeherder/model/derived/jobs.py#L1106-1106

Or was there a reason to keep it?

File: treeherder/webapp/api/resultset.py [hunk 2]
```diff
@@ -152,9 +89,6 @@ def list(self, request, project, jm):
             'results': results,
```
Comment (camd) [inline treeherder/webapp/api/resultset.py]: If you don't have any records in a repo, then ``for rs in objs`` will never initialize ``results`` and it'll die with a 503 on this endpoint.  Should perhaps just initialize ``results = {}`` before that for-loop.

**Comment by camd** [comment]
@maurodoglio - Aside from my few little things, this looks great! Good to merge whenever you're ready.  Now on to the UI PR.  :)

File: treeherder/webapp/api/resultset.py (hunk 2 above)
Comment (maurodoglio) [inline treeherder/webapp/api/resultset.py]: It was actually mis-indented :stuck_out_tongue_winking_eye:

**Comment by edmorley** [comment]
I don't think we should be moving the order mappings from the service to the UI - we have bugs filed for moving more into the service, rather than the other way around (eg bug 1066391). By keeping the mappings and order in the service, other consumers can choose if they want to use the order we provide (for consistency with the UI), or else do their own thing anyway.

Could you also add a multi-line commit message explaining the changes and the reason for them? (Will help with reviewing and also future git archaeology) :-)

File: tests/webapp/api/test_jobs_api.py [hunk 3]
```diff
@@ -53,6 +54,7 @@ def test_job_list(webapp, eleven_jobs_processed, jm):
         "last_modified"
     ]
     for job in jobs:
+        print job.keys()
```
Comment (edmorley) [inline tests/webapp/api/test_jobs_api.py]: Leftover print

**Pull request merged**